### PR TITLE
Dashboard CD to GCP env

### PR DESCRIPTION
**Pull request by acarnagey**
Attempt to passing env to google storage artifact

### CODE REVIEW

**Review by ipbyrne — APPROVED** [review]

File: packages/dashboard/pages/create.tsx [hunk 1]
```diff
@@ -56,6 +56,7 @@ const Resolver: NextPage = () => {
     const didResolution = await response.json();
     wallet.add(didResolution);
     localStorage.setItem('sidetree.wallet', JSON.stringify(wallet));
+    setWallet(wallet);
```
Comment (OR13) [inline packages/dashboard/pages/create.tsx:59]: change unrelated?

File: .github/workflows/cd-dashboard.yml [hunk 2]
```diff
@@ -45,6 +47,25 @@ jobs:
       - name: Build and Push Container
         working-directory: ./packages/dashboard
         run: |
+          cat << EOF > .env
```
Comment (OR13) [inline .github/workflows/cd-dashboard.yml:50]: you create this file here... but where do you use it?

File: packages/dashboard/pages/create.tsx (hunk 1 above)
Comment (acarnagey) [reply packages/dashboard/pages/create.tsx:59]: Not related but it should show the did it created after the API call

File: .github/workflows/cd-dashboard.yml [hunk 3]
```diff
@@ -57,6 +78,6 @@ jobs:
...
+            --set-env-vars NEXT_PUBLIC_SIDETREE_METHOD="${{ env.NEXT_PUBLIC_SIDETREE_METHOD }}",NEXT_PUBLIC_OPERATOR="${{ env.NEXT_PUBLIC_OPERATOR }}",NEXT_PUBLIC_METHOD="${{ env.NEXT_PUBLIC_METHOD }}",NEXT_PUBLIC_DESCRIPTION="${{ env.NEXT_PUBLIC_DESCRIPTION }}",NEXT_PUBLIC_LOGO_TEXT_DARK="${{ env.NEXT_PUBLIC_LOGO_TEXT_DARK }}",NEXT_PUBLIC_LOGO_TEXT_LIGHT="${{ env.NEXT_PUBLIC_LOGO_TEXT_LIGHT }}",MONGO_DB_CONNECTION_STRING="${{ env.MONGO_DB_CONNECTION_STRING }}",DATABASE_NAME="${{ env.DATABASE_NAME }}",MAX_CONCURRENT_DOWNLOADS="${{ env.MAX_CONCURRENT_DOWNLOADS }}",BATCH_INTERVAL_IN_SECONDS="${{ env.BATCH_INTERVAL_IN_SECONDS }}",OBSERVING_INTERVAL_IN_SECONDS="${{ env.OBSERVING_INTERVAL_IN_SECONDS }}",ELEMENT_CONTENT_ADDRESSABLE_STORE_SERVICE_URI="${{ env.ELEMENT_CONTENT_ADDRESSABLE_STORE_SERVICE_URI }}",ELEMENT_ANCHOR_CONTRACT="${{ env.ELEMENT_ANCHOR_CONTRACT }}",ETHEREUM_RPC_URL="${{ env.ETHEREUM_RPC_URL }}",ETHEREUM_PROVIDER="${{ env.ETHEREUM_PROVIDER }}",ETHEREUM_MNEMONIC="${{ env.ETHEREUM_MNEMONIC }}" \
```
Comment (OR13) [inline .github/workflows/cd-dashboard.yml:81]: These changes do not use the file you are creating.

Comment (acarnagey) [reply .github/workflows/cd-dashboard.yml:81]: it's used for node server environment, the file is used for client when next js builds

**Review by OR13 — CHANGES_REQUESTED** [review]
There has to be a better way. 

We should not be writing github secrets to disk.

I thought we proved that it was possible to pass values through... 

If we are having trouble reading them, its possible that they need to be escaped or encoded.

File: .github/workflows/cd-dashboard.yml (hunk 2 above)
Comment (acarnagey) [reply .github/workflows/cd-dashboard.yml:50]: It's used for gcloud build submit, it references .env that is in gcp cloud bucket cloud build for nextjs client

File: .github/workflows/cd-dashboard.yml (hunk 3 above)
Comment (OR13) [reply .github/workflows/cd-dashboard.yml:81]: but doesn't that mean you are bundling the secrets into the container?

File: .github/workflows/cd-dashboard.yml [hunk 4]
```diff
@@ -45,6 +47,25 @@ jobs:
...
+          MONGO_DB_CONNECTION_STRING=${{ secrets.MONGO_DB_CONNECTION_STRING }}
+          DATABASE_NAME=${{ secrets.DATABASE_NAME }}
+          MAX_CONCURRENT_DOWNLOADS=${{ secrets.MAX_CONCURRENT_DOWNLOADS }}
+          BATCH_INTERVAL_IN_SECONDS=${{ secrets.BATCH_INTERVAL_IN_SECONDS }}
+          OBSERVING_INTERVAL_IN_SECONDS=${{ secrets.OBSERVING_INTERVAL_IN_SECONDS }}
+          ELEMENT_CONTENT_ADDRESSABLE_STORE_SERVICE_URI=${{ secrets.ELEMENT_CONTENT_ADDRESSABLE_STORE_SERVICE_URI }}
+          ELEMENT_ANCHOR_CONTRACT=${{ secrets.ELEMENT_ANCHOR_CONTRACT }}
+          ETHEREUM_RPC_URL=${{ secrets.ETHEREUM_RPC_URL }}
+          ETHEREUM_PROVIDER=${{ secrets.ETHEREUM_PROVIDER }}
+          ETHEREUM_MNEMONIC=${{ secrets.ETHEREUM_MNEMONIC }}
```
Comment (OR13) [inline .github/workflows/cd-dashboard.yml:66]: why would next-js need to access a mnemonic to build?

**Pull request closed without merging**